tests-web/www/js/main.js

```diff
@@ -240,59 +181,60 @@ class TestCase {
-          "Geometry (reimport)",
-          makePlainGeoJsonLayer(network.toGeojsonPlain())
-        )
+function importOSM(groupName, app, osmXML, drivingSide, addOSMLayer) {
+  try {
+    const network = new JsStreetNetwork(osmXML, {
+      driving_side: drivingSide,
+      debug_each_step: app.importSettings.debugEachStep,
+      dual_carriageway_experiment: app.importSettings.dualCarriagewayExperiment,
+    });
+    var group = new LayerGroup(groupName, app.map);
+    if (addOSMLayer) {
+      group.addLayer("OSM", makeOsmLayer(osmXML), { enabled: false });
+    }
+    group.addLayer("Geometry", makePlainGeoJsonLayer(network.toGeojsonPlain()));
+    group.addLayer(
+      "Lane polygons",
+      makeLanePolygonLayer(network.toLanePolygonsGeojson())
+    );
+    group.addLayer(
+      "Lane markings",
+      makeLaneMarkingsLayer(network.toLaneMarkingsGeojson())
+    );
+    // TODO Graphviz hits `ReferenceError: can't access lexical declaration 'graph' before initialization`
+
+    const numDebugSteps = network.getDebugSteps().length;
+    group.setEnabled(numDebugSteps == 0);
+    app.layers.addGroup(group);
+
+    var i = 0;
+    for (const step of network.getDebugSteps()) {
+      i++;
+      var group = new LayerGroup(`Step ${i}: ${step.getLabel()}`, app.map);
+      group.addLayer(
+        "Geometry",
+        makePlainGeoJsonLayer(step.getNetwork().toGeojsonPlain())
       );
-      this.layers.push(
-        this.app.addLayer(
-          "Lane polygons (reimport)",
-          makeLanePolygonLayer(network.toLanePolygonsGeojson())
-        )
+      group.addLayer(
+        "Lane polygons",
+        makeLanePolygonLayer(step.getNetwork().toLanePolygonsGeojson())
       );
-      this.layers.push(
-        this.app.addLayer(
-          "Lane markings (reimport)",
-          makeLaneMarkingsLayer(network.toLaneMarkingsGeojson())
-        )
+      group.addLayer(
+        "Lane markings",
+        makeLaneMarkingsLayer(step.getNetwork().toLaneMarkingsGeojson())
       );
-      /*this.layers.push(
-        this.app.addLayer(
-          "Network (reimport)",
-          await makeDotLayer(network.toGraphviz(), { bounds: this.bounds })
-        )
-      );*/
-      for (const step of network.getDebugSteps()) {
-        this.layers.push(
-          this.app.addLayer(
-            `${step.getLabel()}: geometry`,
-            makePlainGeoJsonLayer(step.getNetwork().toGeojsonPlain())
-          )
-        );
-        const debugGeojson = step.toDebugGeojson();
-        if (debugGeojson) {
-          this.layers.push(
-            this.app.addLayer(
-              `${step.getLabel()}: debug`,
-              makeDebugLayer(debugGeojson)
-            )
-          );
-        }
+
+      const debugGeojson = step.toDebugGeojson();
+      if (debugGeojson) {
+        group.addLayer("Debug", makeDebugLayer(debugGeojson));
       }
-    } catch (err) {
-      window.alert(`Reimport failed: ${err}`);
+      group.setEnabled(i == numDebugSteps);
```

Comment:
Only show the last step by default. This is temporary behavior until I make some kind of "scroll through this sequence of groups" control.